apps/files/src/components/DragAndDropNotice.vue

```diff
@@ -186,16 +203,21 @@ export default defineComponent({
 
 			if (lastUpload !== undefined) {
 				logger.debug('Scrolling to last upload in current folder', { lastUpload })
-				this.$router.push({
-					...this.$route,
+				const location: RawLocation = {
+					path: this.$route.path,
+					// Keep params but change file id
 					params: {
-						view: this.$route.params?.view ?? 'files',
-						fileid: parseInt(lastUpload.response!.headers['oc-fileid']),
+						...this.$route.params,
+						fileid: String(lastUpload.response!.headers['oc-fileid']),
 					},
-				})
+				}
+				// Remove open file from query
+				delete location.query?.openfile
```

Comment:
You're not restoring the previous `query` anymore, right? Currently, `location.query?` will always be undefined.

Reply:
fixed it! Now this is always true so can remove the optional chaining:

```suggestion
				delete location.query.openfile
```